rfc/rfc-59/rfc-59.md

```diff
@@ -0,0 +1,285 @@
+3. Add some configuration options (e.g. set which fields are event-time fields and thus need to be verified) in order to faciliate developers to use this feature conveniently. Also, some miscellaneous of trivial adjustment of Hudi logic workflow to make Hudi easily invokes the modifications and new-added functions described above in order to achieve the feature when developer set those options.
+
+We will illustrate them despectively in detail in the following section.
+
+For better understanding, we provided a general workflow to show how the entire process works.
+
+![a general workflow to show how the entire process works](./imgs/Hudi-RFC59-03.png)
+
+### Core part: the Payload class
+
+This Payload class is the core of the entire new feature. It has two key functions and one crucial property to support the entire feature running properly. Let's describe the two functions first.
+
+Firstly, the `preCombineWithMultipleFields` function helps Hudi to deduplicate records inside a buffered bucket with same primary key(s) by multiple event-time fields and only remain the latest record whose every event-time field is latest. 
+
+This function would be invoked in `hudi-client/hudi-flink-client/src/main/java/org/apache/hudi/table/action/commit/FlinkWriteHelper.java` # `deduplicateRecordsWithMultipleCombineFields` to deduplicate record. The deduplication process will also be described later.
+
+Secondly, the `needUpdatingPersistedRecordWithMultipleFields` functions helps Hudi to verify whether all event-time fields are the latest compared with the record persisted on the Disk when sinking the deduplicated record to the Disk before real write operation conducts.
+
+The high-level entire design of this Payload class is shown here:
+
+```java
+
+// hudi-common/src/main/java/org/apache/hudi/common/model/VerifyMultipleLatestEventTimesPayload.java
+public class VerifyMultipleLatestEventTimesPayload extends BaseAvroPayload implements HoodieRecordPayload<VerifyMultipleLatestEventTimesPayload> {
+
+  // A HashMap in order to store the relationship between event-time fieldnames and values
+  private HashMap<String, Comparable<?>> eventTimeFieldNamesAndValues;
+
+  // A string in order to save the eventime-time fieldname of orderingVal. 
+  // This is a compromise designed to be compatible with the current single event-time (property 'orderingVal' provided by BaseAvroPayload). 
+  private String orderingValFieldName;
+
+  // Construction compatible with the current workflow
+  public VerifyMultipleLatestEventTimesPayload(GenericRecord record, Comparable orderingVal, String orderingValFieldName) ;
+
+  // Construction for the multiple event-time latest verification
+  public VerifyMultipleLatestEventTimesPayload(GenericRecord record, HashMap<String, Comparable<?>> eventTimeFieldNamesAndValues) ;

```

Comment:
can you help me understand something. when reading a record from storage (already stored in hudi), how would we get all values for multiple ordering fields? do we deserialize the entire record and then fetch the respective values is it? can you check the flow and ensure it happens.